sycl/doc/extensions/proposed/sycl_ext_oneapi_composite_device.asciidoc

```diff
@@ -0,0 +1,290 @@
+= sycl_ext_oneapi_composite_device
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2023-2023 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 8 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Backend support status
+
+The APIs defined in this extension are only useful when using the Level Zero
+backend, and they are only useful when the Level Zero environment variable
+`ZE_FLAT_DEVICE_HIERARCHY=COMBINED` is set.  The APIs may be called even when
+using other backends, but they will return an empty list of composite devices.
+
+
+== Overview
```

Comment:
I'm not sure, but I wonder if it's worth making this overview slightly more general, since there's no reason that this composite/component behavior couldn't also be applied to other devices from other backends (e.g., CPU devices).
I'm happy to leave this for now if you'd like, and revisit if the scope of the extension is expanded.

Reply:
I think I'd rather wait.  I don't want to specifically state that CPU devices are affected by this extension unless we actually implement that.  I suspect this extension will most likely be implemented first for GPU only.

If we add CPU support later, I think we can generalize the overview and note the behavior with CPU devices.